design/ceph-openshift-fencing-mitigation.md

```diff
@@ -0,0 +1,52 @@
+# Handling Openshift Fencing through managed MachineDisruptionBudgets
+
+## Goals
+
+- Use the MachineDisruptionBudget to ensure that OCP 4.x style fencing does not cause data unavailability and loss.
+
+## What is fencing in OCP 4.x?
+
+Openshift uses `Machines` and `MachineSets` from the [cluster-api](https://github.com/kubernetes-sigs/cluster-api) to dynamically provisions nodes. Fencing is a remediation method that reboots/deletes `Machine` CRDs to solve problems with automatically provisioned nodes.
+
+Once [MachineHealthCheck controller](https://github.com/openshift/machine-api-operator#machine-healthcheck-controller) detects that a node is `NotReady` (or some other configured condition), it will remove the associated `Machine` which will cause the node to be deleted. The `MachineSet` controller will then replace the `Machine` via the machine-api. The exception is on baremetal platforms where fencing will reboot the underlying `BareMetalHost` object instead of deleting the `Machine`.
+
+
+## Why can't we use `PodDisruptionBudget`?
+
+Fencing does not use the eviction api. It is for `Machine`s and not `Pod`s.
```

Comment:
It's a bit more than that, as discussed we can mention why OSDs are not a good fit for pdb.

Reply:
Actually PDBs for OSDs are in progress: https://github.com/rook/rook/blob/master/design/ceph-managed-disruptionbudgets.md

Reply:
I can imagine cloud cases being different, but on metal is it not a problem that PDBs can only enforce that the OSD is running before another one is stopped, not that the replica(s) the OSD manages are up to date?